controllers/configurationpolicy_controller.go

```diff
@@ -1030,11 +1031,34 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 						return
 					}
 
+					if resolvedTemplate.HasSensitiveData {
+						for i := range objTemps {
+							if objTemps[i].RecordDiff == "" {
+								log.V(1).Info(
+									"Not automatically turning on recordDiff due to templates interacting with "+
+										"sensitive data",
+									"objectTemplateIndex", i,
+								)
+
+								objTemps[i].RecordDiff = policyv1.RecordDiffSensored
+							}
+						}
+					}
+
 					plc.Spec.ObjectTemplates = objTemps
 
 					break
 				}
 
+				if plc.Spec.ObjectTemplates[i].RecordDiff == "" && resolvedTemplate.HasSensitiveData {
```

Comment:
When a case of resolvedTemplate.HasSensitiveData falls here...  I think all HasSensitiveData cases falls to line 1034.